test/WebJobs.Script.Tests/FunctionMetadataProviderTests.cs

```diff
@@ -43,6 +38,31 @@ public void ReadFunctionMetadata_Succeeds()
             Assert.True(AreRequiredMetricsEmitted(_testMetricsLogger));
         }
 
+        [Fact]
+        public void ReadFunctionMetadata_With_Retry_Succeeds()
+        {
+            string functionsPath = Path.Combine(Environment.CurrentDirectory, @"..\..\..\..\..\sample\node-retry");
+            _scriptApplicationHostOptions.ScriptPath = functionsPath;
+            var optionsMonitor = TestHelpers.CreateOptionsMonitor(_scriptApplicationHostOptions);
+            var metadataProvider = new FunctionMetadataProvider(optionsMonitor, NullLogger<FunctionMetadataProvider>.Instance, _testMetricsLogger);
+            var workerConfigs = TestHelpers.GetTestWorkerConfigs();
+            var functionMetadatas = metadataProvider.GetFunctionMetadata(workerConfigs, false);
+
+            Assert.Equal(2, functionMetadatas.Length);
+
+            var functionMetadataWithRetry = functionMetadatas.Where(f => f.Name.Contains("retry", StringComparison.OrdinalIgnoreCase));
+            Assert.Single(functionMetadataWithRetry);
+            var retry = functionMetadataWithRetry.FirstOrDefault().FunctionRetry;
+            Assert.NotNull(retry);
+            Assert.Equal(RetryStrategy.FixedDelay, retry.Strategy);
+            Assert.Equal(2, retry.MaxRetryCount);
+            Assert.Equal(TimeSpan.Parse("00:00:10"), TimeSpan.FromSeconds(10));
+
+            var functionMetadata = functionMetadatas.Where(f => !f.Name.Contains("retry", StringComparison.OrdinalIgnoreCase));
```

Comment:
We're not verifying the host.json retry parsing anywhere are we?

Reply:
nope. I think validation built into webjobs sdk is enough.

Reply:
Added validation - creating attributes earlier earlier so it fails sooner